### PR TITLE
use XPRSmipoptimize and XPRSlpoptimize instead of XPRSminim and XPRSmaxim

### PR DESCRIPTION
**Pull request by sgatto**
`XPRSminim` and `XPRSmaxim`are deprecated see [xpress doc](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSmaxim.html)
I propose to use `XPRSlpoptimize` and `XPRSmipoptimize` as suggested by the doc

### CODE REVIEW

File: ortools/linear_solver/xpress_interface.cc [hunk 1]
```diff
@@ -1912,16 +1912,10 @@ MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
 
   int xpress_stat = 0;
   if (mMip) {
-    if (this->maximize_)
-      status = XPRSmaxim(mLp, "g");
-    else
-      status = XPRSminim(mLp, "g");
+    status = XPRSmipoptimize(mLp, "g");
```
Comment (flomnes) [inline ortools/linear_solver/xpress_interface.cc:1915]: Flag "g" is not supported by `XPRSmipoptimize` and `XPRSlpoptimize`.

It means
> g the MIP problem will be solved, calling [XPRSmipoptimize](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSmipoptimize.html) ( [MIPOPTIMIZE](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSmipoptimize.html)).

It seems that no flag should be given.

See
https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSmaxim.html
https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSlpoptimize.html
https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSmipoptimize.html

Comment (pet-mit) [reply ortools/linear_solver/xpress_interface.cc:1915]: I agree with @flomnes , keep the flag empty (""), hoping that XPRSmipoptimize would use the DEFAULTALG to solve the initial relaxation (not explicited in documentation unfortunately)

File: ortools/linear_solver/xpress_interface.cc [hunk 2]
```diff
@@ -1912,16 +1912,10 @@ MPSolver::ResultStatus XpressInterface::Solve(MPSolverParameters const& param) {
...
     XPRSgetintattrib(mLp, XPRS_MIPSTATUS, &xpress_stat);
   } else {
-    if (this->maximize_)
-      status = XPRSmaxim(mLp, "");
-    else
-      status = XPRSminim(mLp, "");
+    status = XPRSlpoptimize(mLp, "g");
```
Comment (pet-mit) [inline ortools/linear_solver/xpress_interface.cc:1918]: you should definitely keep the flags empty here, to keep the same behaviour as before (see doc)
```suggestion
    status = XPRSlpoptimize(mLp, "");
```

**Review by pet-mit — CHANGES_REQUESTED** [review]
**very important**: you should also call XPRSchgobjsense before calling XPRSmipoptimize or XPRSlpoptimize, it is currently only called in the constructor (the user can change the optimization sense at any time by using MPObjective::SetMinimization/Maximization)

File: ortools/linear_solver/xpress_interface.cc (hunk 2 above)
Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc:1918]: As far as I understand, 
`MPObjective::SetMinimization` and `MPObjective::SetMaximisation`
call 
`MPObjective::SetOptimizationDirection`
that calls
`interface_->SetOptimizationDirection`
that is overridden by
```c++
void XpressInterface::SetOptimizationDirection(bool maximize) {
  InvalidateSolutionSynchronization();
  XPRSchgobjsense(mLp, maximize ? XPRS_OBJ_MAXIMIZE : XPRS_OBJ_MINIMIZE);
}
```
So for me there is no need to call `XPRSchgobjsense` again.
Can you confirm @flomnes @pet-mit?

File: ortools/linear_solver/xpress_interface.cc (hunk 1 above)
Comment (sgatto) [reply ortools/linear_solver/xpress_interface.cc]: done

File: ortools/linear_solver/xpress_interface.cc (hunk 2 above)
Comment (pet-mit) [reply ortools/linear_solver/xpress_interface.cc]: you're right, sorry, my ctrl+f did not work :p

**Review by pet-mit — APPROVED** [review]

**Pull request merged**